x/evm/keeper/keeper.go

```diff
@@ -17,6 +17,11 @@ import (
 	"github.com/tharsis/ethermint/x/evm/types"
 )
 
+type cachedContext struct {
+	ctx    sdk.Context
+	commit func()
+}
+
```

Comment:
I think this is relevant enough to have its dedicated file in `keeper/context.go`. Also, consider implementing the slice as a stack data structure?

Reply:
how about 
cachedContext-> CachedContext

Reply:
I used the name `ContextStack`, also the module is named after that.